src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -308,66 +308,71 @@ public static void makeGenotypeCall(final int ploidy,
-                        .mapToObj(GATKVariantContextUtils::formatGP).toArray());
-                gb.attribute(GATKVCFConstants.GENOTYPE_PRIOR_KEY, Arrays.stream(log10Priors)
-                        .map(v -> v == 0.0 ? 0.0 : v * -10)
-                        .mapToObj(GATKVariantContextUtils::formatGP).toArray());
-                // Set the GQ accordingly
-                final int maxPosteriorIndex = MathUtils.maxElementIndex(log10Posteriors);
-                if ( allelesToUse.size() > 0 ) {
-                    gb.log10PError(getGQLog10FromPosteriors(maxPosteriorIndex, normalizedLog10Posteriors));
+            case BEST_MATCH_TO_ORIGINAL -> {
+                Utils.nonNull(originalGT,  "original GT cannot be null if assignmentMethod is " + GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL);
+                gb.alleles(bestMatchToOriginalGT(allelesToUse, originalGT));
+            }
+            case USE_POSTERIOR_PROBABILITIES -> {
+                if (gpc == null) {
+                    throw new GATKException("cannot uses posteriors without an genotype prior calculator present");
+                } else {
+                    // Calculate posteriors.
+                    final double[] log10Priors = gpc.getLog10Priors(ploidy, allelesToUse);
+                    final double[] log10Posteriors = MathUtils.ebeAdd(log10Priors, genotypeLikelihoods);
+                    final double[] normalizedLog10Posteriors = MathUtils.scaleLogSpaceArrayForNumericalStability(log10Posteriors);
+                    // Update GP and PG annotations:
+                    gb.attribute(VCFConstants.GENOTYPE_POSTERIORS_KEY, Arrays.stream(normalizedLog10Posteriors)
+                            .map(v -> v == 0.0 ? 0.0 : v * -10) // the reason for the == 0.0 is to avoid a signed 0 output "-0.0"
+                            .mapToObj(GATKVariantContextUtils::formatGP).toArray());
+                    gb.attribute(GATKVCFConstants.GENOTYPE_PRIOR_KEY, Arrays.stream(log10Priors)
+                            .map(v -> v == 0.0 ? 0.0 : v * -10)
+                            .mapToObj(GATKVariantContextUtils::formatGP).toArray());
+                    // Set the GQ accordingly
+                    final int maxPosteriorIndex = MathUtils.maxElementIndex(log10Posteriors);
+                    if (!allelesToUse.isEmpty()) {
+                        gb.log10PError(getGQLog10FromPosteriors(maxPosteriorIndex, normalizedLog10Posteriors));
+                    }
+                    // Finally we update the genotype alleles.
+                    gb.alleles(GenotypesCache.get(ploidy, maxPosteriorIndex).asAlleleList(allelesToUse));
                 }
-                // Finally we update the genotype alleles.
-                gb.alleles(GenotypesCache.get(ploidy, maxPosteriorIndex).asAlleleList(allelesToUse));
             }
+            case USE_POSTERIORS_ANNOTATION, DO_NOT_ASSIGN_GENOTYPES-> { } // do nothing in these cases
```

Comment:
The case DO_NOT_ASSIGN_GENOTYPES previously just fell through, now it's at least listed.